artifacts/definitions/Windows/Detection/BinaryHunter.yaml

```diff
@@ -0,0 +1,141 @@
+    description: Regex to whitelist in all Authenticode data.
+    default:
+    type: regex
+  - name: PEInformationRegex
+    description: Regex to filter for PE information. e.g VersionInformation, exports etc
+    default: .
+    type: regex
+  - name: PEInformationWhitelistRegex
+    description: Regex to whitelist for PE information. e.g VersionInformation, exports etc
+    default:
+    type: regex
+  - name: DateAfter
+    description: Search for binaries with timestamps after this date. YYYY-MM-DDTmm:hh:ssZ
+    type: timestamp
+  - name: DateBefore
+    description: Search for binaries with timestamps before this date. YYYY-MM-DDTmm:hh:ssZ
+    type: timestamp
+  - name: SizeMax
+    description: Return binaries only under this size in bytes.
+    type: int64
+  - name: SizeMin
+    description: Return binaries only over this size in bytes.
+    type: int64
+  - name: MD5List
+    description: SHA1 hash list to hunt for. New MD5 hash on each line
+    default:
+  - name: SHA1List
+    description: SHA1 hash list to hunt for. New SHA1 hash on each line
+    default:
+  - name: SHA256List
+    description: SHA256 hash list to hunt for. New SHA256 hash on each line
+    default:
+    
+export: |
+  LET Profile = '''[
+    ["FileType", 0, [
+        ["Magic", 0, "String", {
+            "length": 2 
+        }],
+        ["MagicHex",0,"Value",{ 
+            "value": "x=>format(format='% x', args=x.Magic)" 
+        }],
+    ]]]'''
+
+
+sources:
+  - query: |
+      -- time testing
+      LET time_test(stamp) =
+            if(condition= DateBefore AND DateAfter,
+                then= stamp < DateBefore AND stamp > DateAfter,
+                else=
+            if(condition=DateBefore,
+                then= stamp < DateBefore,
+                else=
+            if(condition= DateAfter,
+                then= stamp > DateAfter,
+                else= True
+            )))
+            
+      SELECT
+        dict(FullPath=FullPath,Name=Name,Size=Size,
+            Timestamps=dict(Mtime=Mtime,Atime=Atime,Ctime=Ctime,Btime=Btime)
+                ) as File,
+        authenticode(filename=FullPath) as Authenticode,
+        parse_binary(filename=FullPath,profile=Profile,struct="FileType").MagicHex as _MagixHex,
+        read_file(filename=FullPath,offset=0,length=2) as _Header,
+        parse_pe(file=FullPath) as PE,
+        hash(path=FullPath) as Hash
+      FROM glob(globs=TargetGlob,accessor=Accessor)
+      WHERE 
+        NOT IsDir AND NOT IsLink AND _MagixHex = '4d 5a'
+        AND if(condition= UnexpectedExtension,
+                then= NOT Name =~ '\.(com|cpl|dll|drv|exe|mui|scr|sfx|sys|winmd)$',
+                else= True)
+        AND if(condition=SizeMax,
+                then=FileSize < atoi(string=SizeMax),
+                else=TRUE)
+            AND if(condition=SizeMin,
+                then=FileSize > atoi(string=SizeMin),
+                else=TRUE)
+            AND
+             ( time_test(stamp=Mtime)
```

Comment:
I would recommend to split this timestamp query into separate queries to avoid doing all these calls if the user doesnt care about times.